### ISSUE TITLE
Improper CLI parsing of settings_set

### ISSUE THREAD

**Issue by kauffj** [lbryio/lbry#930]
Calling `./lbrynet-cli settings_set max_key_fee="zzz"` instead set my `download_directory` to `max_key_fee="zzz"`

```
kauffj@kauffj-T570:~/Downloads$ ./lbrynet-cli settings_set max_key_fee="zzz"
{
  "allowed_origin": "",
  "announce_head_blobs_only": false,
  "api_host": "localhost",
  "api_port": 5279,
  "auto_re_reflect": true,
  "auto_re_reflect_interval": 3600,
  "cache_time": 150,
  "data_dir": "/home/kauffj/.local/share/lbry/lbrynet",
  "data_rate": 0.0001,
  "delete_blobs_on_remove": true,
  "dht_node_port": 4444,
  "disable_max_key_fee": true,
  "download_directory": "max_key_fee=zzz",
  "download_timeout": 180,
  "is_generous_host": true,
  "known_dht_nodes": [
    [
      "lbrynet1.lbry.io",
      4444
    ],
    [
      "lbrynet2.lbry.io",
      4444
    ],
    [
      "lbrynet3.lbry.io",
      4444
    ]
  ],
  "lbryum_wallet_dir": "/home/kauffj/.local/share/lbry/lbryum",
  "max_connections_per_stream": 5,
  "max_key_fee": {
    "amount": null,
    "currency": "LBC"
  },
  "min_info_rate": 0.02,
  "min_valuable_hash_rate": 0.05,
  "min_valuable_info_rate": 0.05,
  "peer_port": 3333,
  "peer_search_timeout": 3,
  "pointtrader_server": "http://127.0.0.1:2424",
  "reflect_uploads": true,
  "reflector_port": 5566,
  "reflector_servers": [
    [
      "reflector.lbry.io",
      5566
    ]
  ],
  "run_reflector_server": false,
  "sd_download_timeout": 3,
  "seek_head_blob_first": true,
  "share_usage_data": true,
  "use_auth_http": false,
  "use_upnp": true,
  "wallet": "lbryum"
}
```

**Comment by kauffj** [comment]
(I understand this is the incorrect syntax and that it interpreted my first argument as a single value rather than a named parameter.)

**Comment by kauffj** [comment]
Two proposed changes:

1. Drop support for positional arguments on the CLI for `settings_set`.
2. Ensure the `download_directory` argument provided to `settings_set` exists before accepting the setting. If it doesn't exist, error rather than accepting the value.

**Comment by KamilPiekutowski** [comment]
It appears that dictionary {u"download_directory":" u"max_key_fee=zzz"") is being passed to settings_set
in Deamon.py.
It can be seen in kwargs below.
        yield self._update_settings(kwargs)

Any ideas why would max_key_fee="zzz" be assigned to download_directory before it gets here?
I also looked at the buffer and it is already in there.

**Comment by rickbatka** [comment]
@KamilPiekutowski I think I'm going to take this on, if that's OK with you!

**Comment by KamilPiekutowski** [comment]
@rickbatka  sure not a problem. I hope I was helpful a little.

**Comment by rickbatka** [comment]
Yeah @KamilPiekutowski I just took a look and it does look like the code in Daemon.py is operating correctly; the dictionary is already broken when it gets there like you said.

I traced it back as far as I could and found that the JSON that hits the server.py _render method is broken:
```
#daemon/auth/server.py
...
def _render(self, request):
        print(request)
...
# this prints {"jsonrpc": "2.0", "params": {"download_directory": "bogussetting"}....
```
I don't know yet exactly where things are going wrong, but the JSON is bad before it gets to server.py - it could be there is some preprocessing of the JSON here on the daemon before it gets to the _render method (unlikely), or more likely, the problem is in the lbrynet-cli client code that creates the JSON request in the first place.

Can you point me to the code for the cli client?

**Comment by rickbatka** [comment]
Never mind, I found it - it is on the cli side. Looks like a docopt change will fix that part. I think I can handle it from here.

**Comment by KamilPiekutowski** [comment]
Let me know, if you need my help.

**Comment by rickbatka** [comment]
@kauffj See my PR above. A few notes and follow-up questions:

1. I disabled positional arguments altogether for settings_set - I agree that it didn't make sense for it to work that way. Besides updating the documentation in docs/cli.md, and putting a note in the changelog, is there anything else I need to do before making a breaking change to the CLI like this?
1. In implementing your second request (validating download_directory before setting it), I found that in some cases, settings_set would fail and instead of throwing the exception and quitting, would log a warning and swallow the exception and continue, acting like it worked when really it didn't. I changed that behavior - it now consistently raises an unhandled exception when it can't parse the setting instead of logging a warning and pretending to succeed. If you don't like that change, it's easy enough for me to set it back to the old way, but it was already doing that in some cases and it seems like you'd want it to consistently fail or succeed, not sometimes-fail and sometimes-warn-and-pretend-to-succeed.

The only difference in behavior in regards to point #2 is that when you issue a bogus settings_set command from the CLI, it will tell you it's bogus instead of silently failing to update the setting and returning the array, making it seem like it worked.
